docs: gno CLI tool sub commands

falondarville opened the pull request:
This PR adds sub commands missing from the`gno` CLI tool documentation. I kept the verbiage in as close alignment with my terminal output to keep consistency. 

Comprehensive changes:
* Add missing sub commands from `gno` CLI tool documentation
* Add usage column to Sub Commands table
* Add `mod` sub commands
* Rename `test` and `transpile` “Options” to “Flags” to clarify input in usage
* Add instruction for users to obtain a comprehensive list of `gno` commands

Note that the `clean` command appears on the `gno` command output in my terminal. However, when I run the command, I get `not a gno module: gno.mod file not found in current or any parent directory`. Has the `clean` command been removed or renamed?

For now, I have not included the `clean` command in documentation, as I cannot verify that it exists through testing. 

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [X] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

deelawn reviewed (commented):
Thanks for adding these.  As it currently exists, the flags for the subcommands are incomplete.  Is there a reason why certain flags were included in this PR while flags for subcommands like `run`, `lint`, `repl`, `doc`, `env`, and `bug` were not added / updated?

falondarville commented:
> Thanks for adding these.  As it currently exists, the flags for the subcommands are incomplete.  Is there a reason why certain flags were included in this PR while flags for subcommands like `run`, `lint`, `repl`, `doc`, `env`, and `bug` were not added / updated?

Thanks for reviewing. I added the subcommands of subcommands that had the standardized table formatting. I wasn't sure if the others were WIP or temporary. I added what was clear to me as someone new to this codebase.

deelawn commented:
Okay with merging this @leohhhn?

zivkovicmilos commented:
Hey @deelawn, please give us a minute to give another round of reviews on this before merging 🙏
